lib/src/components/mixins/has_collidables.dart

```diff
@@ -1,14 +1,77 @@
+import 'package:flutter/cupertino.dart';
+
 import '../../../game.dart';
 import '../../components/mixins/collidable.dart';
 import '../../geometry/collision_detection.dart';
 import '../component.dart';
 
 mixin HasCollidables on BaseGame {
-  final List<Collidable> _collidables = [];
+  // final List<Collidable> _collidables = [];
+  final List<Collidable> _activeCollidables = [];
+  final List<Collidable> _noActiveCollidables = [];
+
+  List<Collidable> showActiveCollidables() => [
```

Comment:
This should use a `UnmodifiableListView` instead, it performance better as we don't have to copy the pointers.